src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/XASdkProject.cs

```diff
@@ -54,25 +48,12 @@ public static void SaveNuGetConfig (string directory)
 			doc.Save (Path.Combine (directory, "NuGet.config"));
 		}
 
-		/// <summary>
-		/// Save a global.json to a directory, with version number to support a local build of  Microsoft.Android.Sdk
-		/// </summary>
-		public static void SaveGlobalJson (string directory)
-		{
-			File.WriteAllText (Path.Combine (directory, "global.json"),
-$@"{{
-    ""msbuild-sdks"": {{
-            ""Microsoft.Android.Sdk"": ""{SdkVersion}""
-    }}
-}}");
-		}
-
 		public string PackageName { get; set; }
 		public string JavaPackageName { get; set; }
 
 		public XASdkProject (string outputType = "Exe")
 		{
-			Sdk = $"Microsoft.Android.Sdk/{SdkVersion}";
```

Comment:
This might be a silly question. With the "old" system this `{SdkVersion}` was used to allow Nuget/MSbuild to find the SDK files in our local Nuget cache. How will this work with the new workload system, can it still look in the local cache? Or will it try to download things from nuget.org. What is going to happen on the bots when it's trying to install a new workload because our tests are requesting it. (I should probably read the spec for the workload stuff).

Do we need to be able to support both types of sdk? I'm thinking from a local development point of view.

Reply:
Workloads won't automatically be downloaded from NuGet.org.  They need to be installed.  The version that is used is determined by the workload manifest, which the workload installation process should install or update.

For testing, you would want to "install" the workload you want to test.  For now this basically means copying some files into the dotnet folder.

Reply:
@dellis1972 I think we can migrate entirely away from the old MSBuild style SDK package. For local development, we are now installing a .NET 5 preview build into `$(AndroidToolchainDirectory )/dotnet`. This PR adds a new `ExtractWorkloadPacks` msbuild target to extract the relevant workload .nupkg files into this new .NET 5 installation path for tests running both locally and in CI to use.

For local dev/test efforts the following steps will prepare, build, pack .nupkgs, and install our workload components into the .NET 5 preview sandbox:

`msbuild /Prepare Xamarin.Android.sln`
`msbuild /t:PackDotNet Xamarin.Android.sln`
or 
`make jenkins`
`msbuild /t:CreateAllPacks,ExtractWorkloadPacks build-tools/create-packs/Microsoft.Android.Sdk.proj`

This works for everything aside from runtime packs at the moment, those will still need to be pulled from _some_ NuGet source. Our tests use this local source by default: https://github.com/xamarin/xamarin-android/blob/a7bda88d05b58efd26c17be3956c7140d05868e1/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/XamarinProject.cs#L94

Eventually the work described in https://github.com/dotnet/sdk/issues/14044 will allow us to install runtime packs into our .NET 5 sandbox and we shouldn't have to use NuGet feeds for anything.

Reply:
@pjcollins wrote:

> This PR adds a new `ExtractWorkloadPacks` msbuild target to extract the relevant workload .nupkg files into this new .NET 5 installation path for tests running both locally and in CI to use.

Does this mean that `$(AndroidToolchainDirectory)\dotnet` will be updated to include a copy of in-tree build artifacts as part of `msbuild /t:PackDotNet`?

If so, is there anything we should to to prevent "non-clean" environments in which `$(AndroidToolchainDirectory)\dotnet` contains "possibly corrupt" or "out-of-sync" file contents due to repeated invocations of `msbuild /t:PackDotNet` over time?

Then there's presumably the unit test scenario: won't `msbuild /t:PackDotNet` and an update of `$(AndroidToolchainDirectory)\dotnet` be required to run `dotnet build`-based unit tests?

I'm slightly concerned about the prospect of "corrupting" local `dotnet` installs which will necessitate nuking the `$(AndroidToolchainDirectory)\dotnet` directory, but I don't really have an alternative idea either. 😞

Reply:
For the dotnet/sdk repo, we keep our testing environment separate.  We have a "stage 0" copy of the .NET SDK which is installed under `<REPO ROOT>\.dotnet`.  When we build the repo it creates a "stage 2" version of the .NET SDK in the output folder.  That's what the tests run against.

We may be able to add some hooks which would allow you to set environment variables to point to local workloads without having to copy them into a dotnet install.

Reply:
> Does this mean that $(AndroidToolchainDirectory)\dotnet will be updated to include a copy of in-tree build artifacts as part of msbuild /t:PackDotNet?

Yes it does. `/t:PackDotNet` is a convenience target for the Windows build + smoke job, and for local dev. It effectively wraps `Build`, `CreateAllPacks`, and `ExtractWorkloadPacks`.  The `/t:CreateAllPacks` target can still be used to create all of our .NET .nupkg files without also installing them into `$(AndroidToolchainDirectory)\dotnet\packs\`.

We have a couple of steps which will clean out `$(AndroidToolchainDirectory)\dotnet\packs` to ensure that there are no stale workloads left behind whenever we try to "install" a new one. 
1) `/t:ExtractWorkloadPacks` will delete all `Microsoft.Android.*` content in `$(AndroidToolchainDirectory)\dotnet\packs`, and 
2) `xaprepare` deletes `Android` packs when ensuring that the correct version of .NET Preview is installed:
https://github.com/xamarin/xamarin-android/blob/master/build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetPreview.cs#L26

> Then there's presumably the unit test scenario: won't msbuild /t:PackDotNet and an update of $(AndroidToolchainDirectory)\dotnet be required to run dotnet build-based unit tests?

For unit tests in CI, we download the .nupkgs artifact produced by the build job and then call the `/t:ExtractWorkloadPacks` to "install" them for our tests.